src/blockdata/script.rs

```diff
@@ -313,12 +316,23 @@ impl Script {
 
     /// Generates P2WPKH-type of scriptPubkey
     pub fn new_v0_wpkh(pubkey_hash: &WPubkeyHash) -> Script {
-        Script::new_witness_program(WitnessVersion::V0, &pubkey_hash.to_vec())
+        Script::new_witness_program(WitnessVersion::V0, &pubkey_hash[..])
     }
 
     /// Generates P2WSH-type of scriptPubkey with a given hash of the redeem script
     pub fn new_v0_wsh(script_hash: &WScriptHash) -> Script {
-        Script::new_witness_program(WitnessVersion::V0, &script_hash.to_vec())
+        Script::new_witness_program(WitnessVersion::V0, &script_hash[..])
+    }
+
+    /// Generates P2TR for key spending path
+    pub fn new_v1_p2tr(output_key: TweakedPublicKey) -> Script {
+        Script::new_witness_program(WitnessVersion::V1, &output_key.serialize())
+    }
+
+    /// Generates P2TR for script spending path using an internal public key and some merkle root.
+    pub fn new_v1_p2tr_script<C: Verification>(secp: &Secp256k1<C>, internal_key: UntweakedPublicKey, merkle_root: TapBranchHash) -> Script {
```

Comment:
In affd45408adc858af9a9cbc03a61ca526a5d41ad
nit: Would prefer the name be `new_v1_p2tr_script_spend`